webpack/assets/javascripts/foreman_toast_notifications.js

```diff
@@ -0,0 +1,30 @@
+import ToastActions from './react_app/actions/ToastNotificationActions';
+import $ from 'jquery';
+
+export function notify(notification) {
+  ToastActions.addNotification(notification);
+}
+
+function importFlashMessagesFromRails() {
+  const notifications = $('#notifications').data().flash;
+
+  notifications.forEach(([type, message]) => {
+
+    // normalize rails flash names
+    if (type === 'danger') {
+      type = 'error';
+    }
+
+    notify({type, message, sticky: (type !== 'success')});
+  });
+}
+
+// clear all notifications when leaving the page
+$(window).bind('beforeunload', function () {
```

Comment:
Shouldn't this action be fired on the "componentWillUnmount" lifecycle event of the notification component?